### PR TITLE
Added Perl package in Knowledge base 

### LINKED ISSUE #1536: Add Perl package hints to documentation
Recently, @mbargull has updated some Perl parts of conda-forge and in the process simplified the way Perl packages can be built. This was already exploited with good effect by @tdegeus to migrate some Perl packages from bioconda.

Adding the simple packaging instructions to the knowledge base will help new maintainers to add Perl packages in a consistent manner.

### PR DESCRIPTION
**Pull request by prachinandi**
PR Checklist:

- [x] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] put any other relevant information below
fixes #1536

### CODE REVIEW

**Comment by prachi237** [comment]
@kathatherine  @beckermr   @zklaus  can u please review the PR. This is not the final one lot many changes are yet to be done, just wanted to know if I am proceeding in the right direction and if I have been following exactly what is being asked to fix in issue #1536.

File: src/maintainer/knowledge_base.rst [hunk 1]
```diff
@@ -1536,3 +1536,37 @@ The tl;dr here is that conda sorts as follows:
 So make sure that you **tag** your package in such a way that the package name
 that conda-build spits out will sort the package uploaded with an ``rc`` label
 higher than the package uploaded with the ``dev`` label.
+
+Perl package
+==================
+
+The most commonly used build system for perl packages is ``ExtUtils::MakeMaker``. 
+For the archetypical packaging of packages of this form, you can have a look at ``perl-file-which``'s recipe.
```
Comment (BastianZim) [inline src/maintainer/knowledge_base.rst:1544]: Super small comment: Maybe add a link to the package here?

Comment (prachi237) [reply src/maintainer/knowledge_base.rst:1544]: > Super small comment: Maybe add a link to the package here?

Sure sir! Will do that as well

Comment (prachi237) [reply src/maintainer/knowledge_base.rst:1544]: @BastianZim  from where can I get the package link?😅

Comment (BastianZim) [reply src/maintainer/knowledge_base.rst:1544]: https://github.com/conda-forge/perl-file-which-feedstock

All of conda-forge‘s feedstocks are listed at https://conda-forge.org/feedstock-outputs/

**Comment by zklaus** [comment]
Great start! I think it would also be good to leave a pointer to this section in [the getting started section](https://github.com/conda-forge/conda-forge.github.io/blob/main/src/maintainer/adding_pkgs.rst#getting-started).

I am a bit on the fence. Perhaps it's actually better not to include an example package here, but rather explain the full meta.yaml file. I think in any case, it would be good to add the bit about core, vendor, and site parts of Perl installations so maintainers can get an understanding of this even when not using MakeMaker.

Finally, the line under the section title should have the same length as the title itself.

File: src/maintainer/knowledge_base.rst [hunk 2]
```diff
@@ -1536,3 +1536,37 @@ The tl;dr here is that conda sorts as follows:
...
+
+The steps are as follows: 
+
+* The name is composed of ``perl-`` + the lowered version of the CPAN name.
```
Comment (jaimergp) [inline src/maintainer/knowledge_base.rst:1548]:
```suggestion
* The name is composed of ``perl-`` + the lower-cased version of the CPAN name.
```

File: src/maintainer/knowledge_base.rst [hunk 3]
```diff
@@ -1536,3 +1536,37 @@ The tl;dr here is that conda sorts as follows:
...
+.. code-block::
+
+  build:
+  number: 0
+  noarch: generic
+  script:
+    - perl Makefile.PL INSTALLDIRS=vendor NO_PERLLOCAL=1 NO_PACKLIST=1
+    - make
+    - make test
+    - make install VERBINST=1
```
Comment (jaimergp) [inline src/maintainer/knowledge_base.rst:1563]:
```suggestion
  build:
    number: 0
    noarch: generic
    script:
      - perl Makefile.PL INSTALLDIRS=vendor NO_PERLLOCAL=1 NO_PACKLIST=1
      - make
      - make test
      - make install VERBINST=1
```

File: src/maintainer/knowledge_base.rst [hunk 4]
```diff
@@ -1536,3 +1536,37 @@ The tl;dr here is that conda sorts as follows:
...
+
+``Makefile.PL`` is the crucial one that guarantees installation into the vendor section, as well as non-interference with standard files by the ``NO_PERLLOCAL`` and ``NO_PACKLIST`` switches.
+
+.. note::
+ ``noarch: generic`` should be present only if the package is a pure perl one, i.e. contains no compiled c code or extensions, and that of course more requirements can be necessary.
```
Comment (jaimergp) [inline src/maintainer/knowledge_base.rst:1568]:
```suggestion
 ``noarch: generic`` should be present only if the package is a pure perl one: i.e. contains no compiled C code or extensions, and that of course more requirements can be necessary.
```

> and that of course more requirements can be necessary

What do you mean here exactly?

File: src/maintainer/knowledge_base.rst (hunk 1 above)
Comment (jaimergp) [reply src/maintainer/knowledge_base.rst:1544]: I would actually link to the meta.yaml directly:

https://github.com/conda-forge/perl-file-which-feedstock/blob/main/recipe/meta.yaml

**Pull request closed without merging**